src/ws/process_queue.ts

```diff
@@ -0,0 +1,34 @@
+import { delay } from "../util/utils.ts";
+import { ws } from "./ws.ts";
+
+export async function processQueue(id: number) {
+  const shard = ws.shards.get(id);
+  // If no items or its already processing then exit
+  if (!shard?.queue.length || shard.processingQueue) return;
+
+  shard.processingQueue = true;
+
+  let counter = 0;
+
+  while (shard.queue.length) {
+    if (shard.ws.readyState !== WebSocket.OPEN) {
+    }
```

Comment:
This can block the event loop. So do sth. like this.
```suggestion
    if (shard.ws.readyState !== WebSocket.OPEN) {
      setTimeout(ws.processQueue, 1000);
      break;
    }
```